Purchases/Caching/RCDeviceCache.m

```diff
@@ -99,7 +99,10 @@ - (nullable NSString *)cachedAppUserID {
 }
 
 - (void)cacheAppUserID:(NSString *)appUserID {
-    [self.userDefaults setObject:appUserID forKey:RCAppUserDefaultsKey];
+    @synchronized (self) {
+        self.appUserIDHasBeenSet = YES;
```

Comment:
Would it make sense to move this right after the set in the `userDefaults`? Just in case it fails when calling `setObject`

Reply:
good idea! In practice, I don't think we're actually catching the error anywhere so it'd crash, but if we did add a try / catch this would lead to an inconsistent state. 
Updated.